src/composables/useTransactions.ts

```diff
@@ -166,9 +168,24 @@ function updateTransaction(
   return false;
 }
 
+function isSuccessfulTransaction(transaction: Transaction) {
+  if (transaction.status === 'confirmed') {
+    if (transaction.type === 'order') {
+      return (transaction.receipt as OrderReceipt)?.status === 'fulfilled';
+    } else {
+      return (transaction.receipt as TxReceipt)?.status === 1;
+    }
+  }
+
+  return false;
+}
+
 // Adapted from Uniswap code
 function shouldCheckTx(transaction: Transaction, lastBlockNumber: number) {
-  if (transaction.status === 'confirmed') {
+  if (
+    processedTxs.value.has(transaction.id) ||
+    transaction.status === 'confirmed'
```

Comment:
The `processedTxs` contains pending transactions using the `txListener` - no need to recheck it again since its going to be handled by the `.wait` function.